lib/masamune/abstract_syntax_tree/comment.rb

```diff
@@ -0,0 +1,24 @@
+# TODO: This has only been tested with one-line comments.
+
+module Masamune
+  class AbstractSyntaxTree
+    class Comment < Prism::Comment
+      attr_reader :ast_id
+
+      include NodeHelper
+
+      def initialize(comment_type, comment_location, ast_id)
+        @ast_id = ast_id
+        super(comment_type, comment_location)
+      end
+
+      # TODO: Potentially just grab this from the proper Lex node.
+      # Prism.parse(code).value.slice does not contain any comments that are placed at the end.
```

Comment:
"at the end" of a file or a line? I'm guessing you mean comments at the end of a line?

Reply:
Correct, and now that I'm looking at it again, I think all comments are not included if there's no actual code inbetween the comments.

```ruby
code = <<~CODE
  # First comment
  # Second comment
CODE

parsed_code = Prism.parse(code)
```

This just produces an empty string, because Ripper considers comments to be void statements with the `:void_stmt` identifier:

```
[2] pry(#<TestMasamune>)> parsed_code.value.slice
=> ""
[3] pry(#<TestMasamune>)> 
```

However, if we have a comment in the middle of the code, it shows up:

```ruby
code = <<~CODE
  puts "foo"
  # First comment
  puts "bar"
  # Second comment
CODE

parsed_code = Prism.parse(code)
p parsed_code.value.slice
```

This produces the following:
```
[2] pry(#<TestMasamune>)> parsed_code.value.slice
=> "puts \"foo\"\n# First comment\nputs \"bar\""
```

I wonder if this is a bug?

Reply:
Ah, I'm looking at your other comment about `@source` though, and that DOES print all of the comments:

```ruby
code = <<~CODE
  # First comment
  # Second comment
CODE

parsed_code = Prism.parse(code)
```

```
[1] pry(#<TestMasamune>)> parsed_code.source.source
=> "# First comment\n# Second comment\n"
```

Reply:
`Prism.parse(code)` returns a `Prism::ParseResult` that has an array of comments on it. You probably want to grab those. `Prism.parse(code).comments`.